File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        float max_sum = Float.NEGATIVE_INFINITY;
        int cur_sum = 0;
        for(int i = 0; i < nums.length; i++){
            cur_sum += nums[i];
            if(cur_sum > max_sum) max_sum = cur_sum;
            if(cur_sum < 0) cur_sum = 0;
        }
        return (int)max_sum;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = float('-inf')
        cur_sum = 0
        for i in nums:
            cur_sum += i
            if cur_sum > max_sum:
                max_sum = max(cur_sum, max_sum)
            if cur_sum < 0:
                cur_sum = 0
        return max_sum
